src/sql/workbench/contrib/query/browser/queryActions.ts

```diff
@@ -205,6 +205,7 @@ export class RunQueryAction extends QueryTaskbarAction {
 
 	public override async run(): Promise<void> {
 		if (!this.editor.isSelectionEmpty()) {
+			await this.connectionManagementService.refreshAzureAccountTokenIfNecessary(this.editor.input.uri);
```

Comment:
I initially put this only in runQuery but later found out this check `this.isConnected(this.editor)` in `run` and `runCurrent` will fail and lead to a connection Dialog. 
We could probably remove these two checks since we also do another `this.isConnected(this.editor)` in `runQuery` then both `run` and `runCurrent` get the check result from `runQuery` and go from there. Do you think this is a good idea?

Reply:
Why do the additional checks fail?

If we're duplicating isConnected calls then yeah it sounds reasonable to clean some of those up. But I'd suggest doing that in a separate PR first for more scoped reviewing.

Reply:
`isConnected` checks if stored `ConnectionManagementInfo` has a valid connctionId, which is added via onConnectionComplete calback https://github.com/microsoft/azuredatastudio/blob/cb7b7da0a4bb2b45f0b6092e42fd1ca385938f09/src/sql/platform/connection/common/connectionStatusManager.ts#L138

Thus if there's a pending reconnection (say triggered by listDatabases), the corresponding ConnectionManagementInfo won't have a connctionId. If we run query at this time, `isConnected` check will fail. I saw this happen during my test so I added a refresh call before this check.

I'll keep it as-is for this PR and send a second PR later to remove the duplicate `isConnected` check.